brave/brave-core
# Remove unused custom partner header

Resolves  brave/brave-browser#34971

Relevant Slack discussion: https://bravesoftware.slack.com/archives/C43LW5U2H/p1702943516613989



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Open the network tab of the devtools.
2. Go to https://grammarly.com
3. Confirm that the `X-Brave-Partner` request header is **missing**.

This is what it looked like **prior** to this change:
![Screenshot from 2023-12-19 12-00-44](https://github.com/brave/brave-core/assets/167821/c47c67b2-9d6d-4038-af93-cf8a7d0e1828)
